src/System.Data.SqlClient/src/System/Data/SqlClient/SNI/SslOverTdsStream.cs

```diff
@@ -48,6 +50,51 @@ public void FinishHandshake()
         /// <param name="count">Byte count</param>
         /// <returns>Bytes read</returns>
         public override int Read(byte[] buffer, int offset, int count)
+        {
+            return ReadInternal(buffer, offset, count, CancellationToken.None, false).GetAwaiter().GetResult();
```

Comment:
@stephentoub I have tried to reuse code and use the same method for Read and ReadAsync. Looking forward to any feedback here. 
From what I understand there should be no allocations due to this approach, but looking forward to your review.